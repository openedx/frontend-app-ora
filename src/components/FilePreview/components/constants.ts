import {
  PDFRenderer,
  ImageRenderer,
  TXTRenderer,
} from './FileRenderer/BaseRenderers';

import messages from './messages';

export const errorStatuses = {
  badRequest: 400,
  unauthorized: 401,
  forbidden: 403,
  notFound: 404,
  conflict: 409,
  serverError: 500,
} as const;

export const fileTypes = {
  pdf: 'pdf',
  jpg: 'jpg',
  jpeg: 'jpeg',
  png: 'png',
  bmp: 'bmp',
  txt: 'txt',
  gif: 'gif',
  jfif: 'jfif',
  pjpeg: 'pjpeg',
  pjp: 'pjp',
  svg: 'svg',
} as const;

export const errorMessages = {
  [errorStatuses.notFound]: messages.fileNotFoundError,
  [errorStatuses.serverError]: messages.unknownError,
};

/**
 * Config data
 */
export const renderers = {
  [fileTypes.pdf]: PDFRenderer,
  [fileTypes.jpg]: ImageRenderer,
  [fileTypes.jpeg]: ImageRenderer,
  [fileTypes.bmp]: ImageRenderer,
  [fileTypes.png]: ImageRenderer,
  [fileTypes.txt]: TXTRenderer,
  [fileTypes.gif]: ImageRenderer,
  [fileTypes.jfif]: ImageRenderer,
  [fileTypes.pjpeg]: ImageRenderer,
  [fileTypes.pjp]: ImageRenderer,
  [fileTypes.svg]: ImageRenderer,
};

export const supportedTypes = Object.keys(renderers);
