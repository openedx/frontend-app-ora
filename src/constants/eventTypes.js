import { StrictDict } from '@edx/react-unit-test-utils';

export default StrictDict({
  modalClose: 'plugin.modal-close',
  modalOpen: 'plugin.modal',
});
