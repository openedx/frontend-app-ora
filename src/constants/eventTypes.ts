export default {
  modalClose: 'plugin.modal-close',
  modalOpen: 'plugin.modal',
} as const;
