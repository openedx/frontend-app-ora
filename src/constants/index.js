export const feedbackRequirement = Object.freeze({
  disabled: 'disabled',
  required: 'required',
  optional: 'optional',
});

export const queryKeys = Object.freeze({
  oraConfig: 'oraConfig',
  pageData: 'pageData',
});

export const MutationStatus = Object.freeze({
  idle: 'idle',
  loading: 'loading',
  error: 'error',
  success: 'success',
});

export const stepStates = Object.freeze({
  inProgress: 'inProgress',
  done: 'done',
  cancelled: 'cancelled',
  closed: 'closed',
  notAvailable: 'notAvailable',
  teamAlreadySubmitted: 'teamAlreadySubmitted',
  needTeam: 'needTeam',
  waiting: 'waiting',
  waitingForPeerGrades: 'waitingForPeerGrades',
  submitted: 'submitted', // ui-only
  trainingValidation: 'trainingValidation', // ui-only
});

export const closedReasons = Object.freeze({
  pastDue: 'pastDue',
  notAvailable: 'notAvailable', // (yet)
});

export const stepNames = Object.freeze({
  xblock: 'xblock',
  xblockStudio: 'xblockStudio',
  xblockPreview: 'xblockPreview',
  submission: 'submission',
  peer: 'peer',
  self: 'self',
  studentTraining: 'studentTraining',
  staff: 'staff',
  done: 'done',
});

export const assessmentSteps = [
  stepNames.studentTraining,
  stepNames.self,
  stepNames.peer,
];

export const routeSteps = Object.freeze({
  xblock: stepNames.xblock,
  xblock_studio: stepNames.xblockStudio,
  xblock_preview: stepNames.xblockPreview,
  submission: stepNames.submission,
  peer_assessment: stepNames.peer,
  self_assessment: stepNames.self,
  student_training: stepNames.studentTraining,
  graded: stepNames.done,
});

export const stepRoutes = Object.freeze(Object.keys(routeSteps).reduce(
  (curr, route) => ({ ...curr, [routeSteps[route]]: route }),
  {},
));

export default { feedbackRequirement, queryKeys, MutationStatus };
