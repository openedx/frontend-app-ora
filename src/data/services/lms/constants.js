import { StrictDict } from '@edx/react-unit-test-utils';

export const feedbackRequirement = StrictDict({
  disabled: 'disabled',
  required: 'required',
  optional: 'optional',
});

export const queryKeys = StrictDict({
  oraConfig: 'oraConfig',
  pageData: 'pageData',
});

export const MutationStatus = StrictDict({
  idle: 'idle',
  loading: 'loading',
  error: 'error',
  success: 'success',
});

export const stepStates = StrictDict({
  inProgress: 'inProgress',
  done: 'done',
  cancelled: 'cancelled',
  closed: 'closed',
  notAvailable: 'notAvailable',
  teamAlreadySubmitted: 'teamAlreadySubmitted',
  needTeam: 'needTeam',
  waiting: 'waiting',
});

export const closedReasons = StrictDict({
  pastDue: 'pastDue',
  notAvailable: 'notAvailable', // (yet)
});

export const stepNames = StrictDict({
  xblock: 'xblock',
  submission: 'submission',
  peer: 'peer',
  self: 'self',
  studentTraining: 'studentTraining',
  staff: 'staff',
  done: 'done',
});

export const routeSteps = StrictDict({
  xblock: stepNames.xblock,
  submission: stepNames.submission,
  peer_assessment: stepNames.peer,
  self_assessment: stepNames.self,
  student_training: stepNames.studentTraining,
  graded: stepNames.done,
});

export const stepRoutes = StrictDict(Object.keys(routeSteps).reduce(
  (curr, route) => ({ ...curr, [routeSteps[route]]: route }),
  {},
));

export default { feedbackRequirement, queryKeys, MutationStatus };
