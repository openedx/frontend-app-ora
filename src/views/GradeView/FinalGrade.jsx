import React from 'react';

import { useIntl } from '@edx/frontend-platform/i18n';

import {
  useAssessmentData,
  useStepInfo,
} from 'hooks/app';
import InfoPopover from 'components/InfoPopover';
import ReadOnlyAssessment from 'components/Assessment/ReadonlyAssessment';
import messages, { labelMessages } from './messages';

const FinalGrade = () => {
  const { formatMessage } = useIntl();
  const { effectiveAssessmentType, ...assessments } = useAssessmentData();
  const stepInfo = useStepInfo();

  const loadStepData = (step) => ({
    ...assessments[step],
    key: step,
    step,
    stepLabel: formatMessage(labelMessages[step]),
  });

  const effectiveAssessment = loadStepData(effectiveAssessmentType);
  const finalStepScore = effectiveAssessment?.stepScore;

  const extraGrades = Object.keys(assessments)
    .filter((type) => !!stepInfo[type] && type !== effectiveAssessmentType)
    .map(loadStepData);

  const renderAssessment = (stepData, defaultOpen = false) => (
    <ReadOnlyAssessment {...stepData} defaultOpen={defaultOpen} />
  );

  return (
    <div>
      <h1>
        {formatMessage(messages.yourFinalGrade, finalStepScore)}
        <InfoPopover>
          <p>
            {effectiveAssessmentType === 'peer'
              ? formatMessage(messages.peerAsFinalGradeInfo)
              : formatMessage(messages.finalGradeInfo, { step: effectiveAssessmentType })}
          </p>
        </InfoPopover>
      </h1>
      <div className="py-2">
        {renderAssessment(effectiveAssessment, true)}
      </div>
      <div className="my-2" />
      {extraGrades.length > 0 && (
        <h3>
          {formatMessage(messages.unweightedGrades)}
          <InfoPopover>
            <p>{formatMessage(messages.unweightedGradesInfo)}</p>
          </InfoPopover>
        </h3>
      )}
      {extraGrades.map((assessment) => renderAssessment(assessment, false))}
    </div>
  );
};

FinalGrade.defaultProps = {};
FinalGrade.propTypes = {};

export default FinalGrade;
