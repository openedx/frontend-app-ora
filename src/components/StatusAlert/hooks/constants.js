import { CheckCircle, Info, WarningFilled } from '@edx/paragon/icons';

import { stepStates } from 'constants/index';

export const alertTypes = {
  success: { variant: 'success', icon: CheckCircle },
  danger: { variant: 'danger', icon: Info },
  warning: { variant: 'warning', icon: WarningFilled },
  light: { variant: 'light', icon: null },
  dark: { variant: 'dark', icon: null },
};

export const alertMap = {
  [stepStates.done]: alertTypes.success,
  [stepStates.submitted]: alertTypes.success,
  [stepStates.closed]: alertTypes.danger,
  [stepStates.teamAlreadySubmitted]: alertTypes.warning,
  [stepStates.needTeam]: alertTypes.warning,
  [stepStates.waiting]: alertTypes.warning,
  [stepStates.cancelled]: alertTypes.warning,
  [stepStates.notAvailable]: alertTypes.light,
  [stepStates.inProgress]: alertTypes.dark,
  [stepStates.waitingForPeerGrades]: alertTypes.warning,
};
