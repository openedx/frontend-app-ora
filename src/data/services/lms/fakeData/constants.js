import { StrictDict } from '@edx/react-unit-test-utils';
import { stepNames } from 'data/services/lms/constants';

export const viewKeys = StrictDict({
  xblock: 'xblock',
  submission: 'submission',
  studentTraining: 'student_training',
  self: 'self_assessment',
  peer: 'peer_assessment',
  done: 'graded',
});

export const progressKeys = StrictDict({
  cancelledDuringSubmission: 'cancelled_during_submission',
  cancelledDuringStudentTraining: 'cancelled_during_student_training',
  cancelledDuringSelf: 'cancelled_during_self',
  cancelledDuringPeer: 'cancelled_during_peer',
  cancelledDuringStaff: 'cancelled_during_staff',
  submissionEarly: 'submission_early',
  submissionClosed: 'submission_closed',
  submissionTeamAlreadySubmitted: 'submission_team_already_submitted',
  submissionNeedTeam: 'submission_need_team',
  submissionUnsaved: 'submission_unsaved',
  submissionSaved: 'submission_saved',
  submissionFinished: 'submission_finished',
  studentTraining: 'student_training',
  studentTrainingValidation: 'student_training_validation',
  studentTrainingPartial: 'student_training_partial',
  studentTrainingFinished: 'student_training_finished',
  selfAssessment: 'self_assessment',
  selfAssessmentLate: 'self_assessment_late',
  selfAssessmentFinished: 'self_assessment_finished',
  peerAssessment: 'peer_assessment',
  peerAssessmentEarly: 'peer_assessment_early',
  peerAssessmentLate: 'peer_assessment_late',
  peerAssessmentWaiting: 'peer_assessment_waiting',
  peerAssessmentFinished: 'peer_assessment_finished',
  staffAfterSubmission: 'staff_after_submission',
  staffAfterSelf: 'staff_after_self',
  staffAfterPeer: 'staff_after_peer',
  graded: 'graded',
  gradedSubmittedOnPreviousTeam: 'graded_submitted_on_previous_team',
});

export const teamStates = [
  progressKeys.gradedSubmittedOnPreviousTeam,
  progressKeys.submissionTeamAlreadySubmitted,
  progressKeys.submissionNeedTeam,
];

export const closedStates = {
  open: { isClosed: false },
  notAvailable: { isClosed: true, closedReason: 'notAvailable' },
  closed: { isClosed: true, closedReason: 'pastDue' },
};

export const stepConfigs = StrictDict({
  all: [stepNames.studentTraining, stepNames.self, stepNames.peer, stepNames.staff],
  self: [stepNames.self],
  peer: [stepNames.peer],
  staff: [stepNames.staff],
  trainingAndSelf: [stepNames.studentTraining, stepNames.self],
  trainingAndPeer: [stepNames.studentTraining, stepNames.peer],
  selfAndStaff: [stepNames.self, stepNames.staff],
});

export const stateStepConfigs = {
  [progressKeys.staffAfterSubmission]: stepConfigs.staff,
  [progressKeys.staffAfterSelf]: stepConfigs.selfAndStaff,
};
