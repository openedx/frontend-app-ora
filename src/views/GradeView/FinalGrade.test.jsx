import { shallow } from '@edx/react-unit-test-utils';

import {
  useAssessmentData,
  useStepInfo,
} from 'hooks/app';
import { stepNames } from 'constants/index';

import FinalGrade from './FinalGrade';

jest.mock('hooks/app', () => ({
  useAssessmentData: jest.fn(),
  useStepInfo: jest.fn(),
}));
jest.mock('components/InfoPopover', () => 'InfoPopover');
jest.mock('components/Assessment/ReadonlyAssessment', () => 'ReadOnlyAssessment');

describe('<FinalGrade />', () => {
  const mockUseAssessmentData = {
    effectiveAssessmentType: stepNames.self,
    [stepNames.self]: {
      stepScore: 1,
    },
    [stepNames.peer]: {
      stepScore: 2,
    },
  };

  it('self and peer grades', () => {
    useAssessmentData.mockReturnValue(mockUseAssessmentData);
    useStepInfo.mockReturnValue({
      [stepNames.self]: true,
      [stepNames.peer]: true,
    });

    const wrapper = shallow(<FinalGrade />);
    expect(wrapper.snapshot).toMatchSnapshot();

    expect(wrapper.instance.findByType('ReadOnlyAssessment')).toHaveLength(2);
  });

  it('peer only', () => {
    useAssessmentData.mockReturnValue({
      effectiveAssessmentType: stepNames.peer,
      [stepNames.peer]: {
        stepScore: 2,
      },
    });
    useStepInfo.mockReturnValue({
      [stepNames.self]: false,
      [stepNames.peer]: true,
    });

    const wrapper = shallow(<FinalGrade />);
    expect(wrapper.snapshot).toMatchSnapshot();

    expect(wrapper.instance.findByType('ReadOnlyAssessment')).toHaveLength(1);
  });
});
