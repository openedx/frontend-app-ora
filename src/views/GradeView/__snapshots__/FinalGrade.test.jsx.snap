// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FinalGrade /> peer only 1`] = `
<div>
  <h1>
    Your final grade: {earned}/{possible}
    <InfoPopover>
      <p>
        Only the required number of peer grades will counted against your final grade. The others are shown, but are not included in your grade calculation
      </p>
    </InfoPopover>
  </h1>
  <div
    className="py-2"
  >
    <ReadOnlyAssessment
      defaultOpen={true}
      key="peer"
      step="peer"
      stepLabel="Peer"
      stepScore={2}
    />
  </div>
  <div
    className="my-2"
  />
</div>
`;

exports[`<FinalGrade /> self and peer grades 1`] = `
<div>
  <h1>
    Your final grade: {earned}/{possible}
    <InfoPopover>
      <p>
        Your grade is based on your self score for this problem. Other assessments don't count towards your final score.
      </p>
    </InfoPopover>
  </h1>
  <div
    className="py-2"
  >
    <ReadOnlyAssessment
      defaultOpen={true}
      key="self"
      step="self"
      stepLabel="Self"
      stepScore={1}
    />
  </div>
  <div
    className="my-2"
  />
  <h3>
    Unweighted grades
    <InfoPopover>
      <p>
        These grades are given to your response. However, these are not used to compute your final grade.
      </p>
    </InfoPopover>
  </h3>
  <ReadOnlyAssessment
    defaultOpen={false}
    key="peer"
    step="peer"
    stepLabel="Peer"
    stepScore={2}
  />
</div>
`;
